engine/table/src/main/java/io/deephaven/engine/table/impl/PushdownResult.java

```diff
@@ -49,30 +64,154 @@ public class PushdownResult implements SafeCloseable {
      */
     private final WritableRowSet maybeMatch;
 
-    private PushdownResult(
-            final WritableRowSet match,
-            final WritableRowSet maybeMatch) {
-        this.match = match;
-        this.maybeMatch = maybeMatch;
+    /**
+     * Constructs a new result with all of {@code input} as {@link #maybeMatch() maybeMatch} and {@code selection}.
+     *
+     * @param input the input
+     * @return the result
+     */
+    public static PushdownResult maybeMatch(@NotNull final RowSet input) {
+        try (final WritableRowSet empty = RowSetFactory.empty()) {
+            return copy(empty, input);
+        }
     }
 
+    /**
+     * Constructs a new result with all of {@code input} as {@link #match() match} and {@code selection}.
+     *
+     * @param input the input
+     * @return the result
+     */
+    public static PushdownResult match(@NotNull final RowSet input) {
+        try (final WritableRowSet empty = RowSetFactory.empty()) {
+            return copy(input, empty);
+        }
+    }
+
+    /**
+     * Constructs a new result with all of {@code input} as {@code noMatch} and {@code selection}.
+     *
+     * @param input the input
+     * @return the result
+     */
+    public static PushdownResult noMatch(@SuppressWarnings("unused") @NotNull final RowSet input) {
```

Comment:
Why have a parameter for input here?

Reply:
You're going to break all the other methods anyway.  I think without some clearer Javadoc it isn't clear what you're supposed to pass in, if anything, here.